### ISSUE TITLE
Generate bloom filters from structured metadata

### ISSUE THREAD

**Issue by tiithansen** [grafana/loki#14280]
**Is your feature request related to a problem? Please describe.**
Currently bloom filters are generated from raw log lines and raw log line can contain data which is never used to filter anything for example if log format is `json` we hardly filter logs based on some `json property` name rather we want to filter only by value that field has.

For example:
`{ "proxy_protocol_addr": "123.123.123.123" }` there is no need to generate bloom for string `proxy_protocol_addr` only `123.123.123.123` should be considered for bloom.

**Describe the solution you'd like**
It would be useful if it would be possible to change the source for bloom generation. Either use raw log lines or structured metadata. Having the possibility to generate blooms from structured metadata could give few benefits.
- Less and smaller blooms which could be split up based on structured metadata field. For example if we have a query `{cluster="abcd"} | proxy_protocol_addr="123.123.123"` leaving last segment as wildcard.
- It could speed up bloom generation
- Basically blooms would become secondary indexes
- By parsing query it can quickly detect which blooms to use for filtering

**Describe alternatives you've considered**
No alternatives

**Comment by chaudum** [comment]
Hi @tiithansen 

As you probably have noticed, this is where we are currently pivoting with Blooms. It turned out that the initial idea of building blooms from log content (while really attractive) was very hard to operate at scale.

You can already try out building blooms from structured metadata when running Loki from `main`.

**Comment by tiithansen** [comment]
> Hi [@tiithansen](https://github.com/tiithansen)
> 
> As you probably have noticed, this is where we are currently pivoting with Blooms. It turned out that the initial idea of building blooms from log content (while really attractive) was very hard to operate at scale.
> 
> You can already try out building blooms from structured metadata when running Loki from `main`.

That was the reason why I closed this task as I already tested it on our Loki PoC. One issue I noticed is that filtering has some issues, it seems to consider too many chunks because of this condition probably https://github.com/grafana/loki/blob/6354deda90a9430856447e27123b3a33fd1b77a0/pkg/storage/bloom/v1/bloom_tester.go#L192.

**Comment by rlex** [comment]
Is that still in main or already part of recent loki releases?
